### PR TITLE
task 4 done

### PR DESCRIPTION
**Pull request by arashmjr**

### CODE REVIEW

File: repository/SaveUserRepository.py [hunk 1]
```diff
@@ -9,13 +9,34 @@ def __init__(self, collection: Collection):
...
     def insert(self, model: SaveUserDomainModel):
-        self.collection.insert_one(model.to_dict())
-
+        result = self.collection.insert_one(model.to_dict())
+        return str(result.inserted_id)
 
     def find_record_by_user_id(self, user_id: str):
         return self.collection.find_one({"user_id": user_id})
 
+    def find(self):
```
Comment (behrad-kzm) [inline repository/SaveUserRepository.py:18]: this naming is not correct

File: repository/UserFollowRepository.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    collection: Collection
+
+    def __init__(self, collection: Collection):
+        self.collection = collection
+
+    def insert(self, model: UserFollowDomainModel):
+        result = self.collection.insert_one(model.to_dict())
+        return str(result.inserted_id)
+
+    def find(self, user_id: str):
```
Comment (behrad-kzm) [inline repository/UserFollowRepository.py:15]: this naming is not correct

File: seviceLayer/core/ServiceProvider.py [hunk 3]
```diff
@@ -1,18 +1,24 @@
...
+from seviceLayer.SaveOrderService import SaveOrderService
+from seviceLayer.UserFollowService import UserFollowService
 
 
 class ServiceProvider:
     repository_provider: RepositoryProvider
-    auth: AuthorizationManager
 
     def __init__(self):
-        self.auth = AuthorizationManager.get_instance()
```
Comment (behrad-kzm) [inline seviceLayer/core/ServiceProvider.py:11]: Why you removed Auth?

File: Domain/models/SaveUserDomainModel.py [hunk 4]
```diff
@@ -9,20 +9,26 @@ class SaveUserDomainModel:
...
         return {"user_id": self.user_id,
                 "name": self.name,
                 "photo_url": self.photo_url,
                 "username": self.username,
                 "token": self.token,
-                "phone_number": self.phone_number
+                "phone_number": self.phone_number,
+                "bio":self.bio,
+                "coin": self.coin
                 }
```
Comment (behrad-kzm) [inline Domain/models/SaveUserDomainModel.py:34]: needs new line after ending your class definition.

File: Domain/models/UserFollowDomainModel.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    def __init__(self, user_id: str, follow_id: str, order_id: str):
+        self.user_id = user_id
+        self.follow_id = follow_id
+        self.order_id = order_id
+
+    def to_dict(self):
+        return {"user_id": self.user_id,
+                "follow_id": self.follow_id,
+                "order_id": self.order_id
+                }
```
Comment (behrad-kzm) [inline Domain/models/UserFollowDomainModel.py:18]: needs new line after ending your class definition.

File: repository/SaveOrderRepository.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+             arr.append(item)
+        return arr
+
+    def find_record_by_user_id(self, user_id: str):
+        return self.collection.find_one({"user_id": user_id})
+
+    def update_isActive_field(self, document: dict):
+        my_query = {"isActive": document['isActive']}
+        new_values = {"$set": {"isActive": False}}
+        return self.collection.update_one(my_query, new_values)
```
Comment (behrad-kzm) [inline repository/SaveOrderRepository.py:38]: needs new line after ending your class definition.

File: repository/SaveUserRepository.py (hunk 1 above)
Comment (behrad-kzm) [reply repository/SaveUserRepository.py:18]: try using get_all instead of this

File: repository/SaveUserRepository.py [hunk 7]
```diff
@@ -9,13 +9,34 @@ def __init__(self, collection: Collection):
...
+        for x in self.collection.find():
+            arr.append(x)
+        return arr
 
     def remove_record(self, user_id:  str):
         return self.collection.delete_one({'user_id': user_id})
 
+    def remove_all(self):
+        delete_all = self.collection.delete_many({})
+        print(delete_all.deleted_count, " documents deleted.")
```
Comment (behrad-kzm) [inline repository/SaveUserRepository.py:29]: remove prints from the code you pushing to the codebase.

**Review by behrad-kzm — COMMENTED** [review]
Please fix the following issues

1.Fix the comments. 
2.Revert the AuthorizationManager changes and add them to the servicases.

File: repository/SaveUserRepository.py (hunk 1 above)
Comment (arashmjr) [reply repository/SaveUserRepository.py]: done

File: repository/UserFollowRepository.py (hunk 2 above)
Comment (arashmjr) [reply repository/UserFollowRepository.py]: done

File: seviceLayer/core/ServiceProvider.py (hunk 3 above)
Comment (arashmjr) [reply seviceLayer/core/ServiceProvider.py:11]: it returned

File: Domain/models/SaveUserDomainModel.py (hunk 4 above)
Comment (arashmjr) [reply Domain/models/SaveUserDomainModel.py]: done

**Comment by arashmjr** [comment]
all bugs are fixed

**Comment by arashmjr** [comment]
task 5 done

**Comment by arashmjr** [comment]
dude check pull request please

**Pull request closed without merging**